plugins/gzip.ts

```diff
@@ -0,0 +1,57 @@
+import { merge } from "../core/utils/object.ts";
+import { Page } from "../core/file.ts";
+import { pageMatches } from "../core/processors.ts";
+import { gzip as compress, Memory } from "../deps/foras.ts";
+
+import type { Extensions } from "../core/utils/path.ts";
+import type Site from "../core/site.ts";
+
+export interface Options {
+  /** The list of extensions this plugin applies to */
+  extensions?: Extensions;
+
+  /**
+   * Level param between 0 and 9 (9 is the smallest but takes the longest to encode)
+   */
+  level?: number;
+}
+
+// Default options
+export const defaults: Options = {
+  extensions: [".html", ".css", ".js", ".mjs", ".svg", ".json", ".xml", ".txt"],
```

Comment:
Maybe the default value should be `*`, to support all files.

Reply:
I think the default here should be fine. Since `*` would also compress images, videos and other binary files. Most binary files have their own compression and gzip is not that well suited for binary/non-text data. What do you think?

Reply:
Fair.
Let's default to these extensions then.